### ISSUE TITLE
`.package.resolved` is not updated

### ISSUE THREAD

**Issue by iphilgood** [tuist/tuist#6725]
### What happened?

Hi guys 👋🏽 

I'm facing an issue, where the `.package.resolved` in the project root is not updated after `tuist generate`. This will result in different package versions defined in the `.package.resolved` in the root directory and the `Package.resolved` in the workspace.

I initially thought that this might be related to #6335 but I think that's not the case. During a deep dive, I noticed that the changes introduced with #6394 doesn't update the `.package.resolved` in the root directory anymore after `tuist generate`.

I think it's a good idea to not put the `.package.resolved` file to the `.gitignore` because this way I can monitor the changes made to packages. Additionally, this way I can make sure, that the specified versions really are the dependencies that get installed, right?

I'm also using Xcode's default integration to define our dependencies and we're a team working on a project and I want to make sure that everybody has the same version of a dependency.

### How do we reproduce it?

1. Run `tuist generate`
2. Bump the version of a package
3. Run `tuist generate` again
4. 💥  The `.package.resolved` on the root still contains the old version, although the `Package.resolved` in the workspace contains the right version

### Error log

Unfortunately, there's no error log.

### macOS version

14.6.1

### Tuist version

4.22.0

### Xcode version

15.4.0

**Comment by fortmarek** [comment]
> I think it's a good idea to not put the .package.resolved file to the .gitignore because this way I can monitor the changes made to packages. Additionally, this way I can make sure, that the specified versions really are the dependencies that get installed, right?

Yes, that's correct. `.package.resolved` shouldn't be put into `.gitignore`.

Are you able to provide a reproducible project? I tried to change a package version in the [ios_app_with_remote_swift_package](https://github.com/tuist/tuist/tree/main/fixtures/ios_app_with_remote_swift_package) and couldn't reproduce the issue.

**Comment by iphilgood** [comment]
Unfortunately, I didn't have time to provide an example yet. I'll try to provide one by the end of the week 💪🏽

**Comment by iphilgood** [comment]
Hi @fortmarek 👋🏽 

I've created a small repo right here: https://github.com/iphilgood/tuist-package-resolved

You can reproduce it with the following steps:

1. Download or clone the repo
2. Run `tuist edit`
3. Uncomment line 11 in `Project.swift`
4. Comment line 12 in `Project.swift`
5. Run `tuist generate`
6. No updated `.package.resolved` on the root level
    1.  Although it should be updated with Swinject version `2.9.1` instead of `2.8.0`

I hope this helps and please let me know if you need more information.

**Comment by jesus-mg-ios** [comment]
It is an error of Xcode 16.1, you can solve it removing all caches of SPM and Xcode, and then forcing to resolve packages from command line

**Comment by iphilgood** [comment]
With the latest tuist version I was able to resolve the issue by deleting the workspace which contained the `Package.resolved` file. After that, tuist created a symlink to the `.package.resolved` in the project root. Now, everything is working as expected 💪🏽 Thank you guys!
